### PR TITLE
Feature/rnp 135

### PR DESCRIPTION
**Pull request by MichielVrins**

### CODE REVIEW

File: ios/Handlers/ResourcesHandler.swift [hunk 1]
```diff
@@ -65,7 +89,7 @@ class ResourceHandler: BridgeToResourceHandlerProtocol {
     private func getHeaders(_ details: Dictionary<String, Any?>) throws -> [String: String] {
         guard let headers = details["headers"] as? [String: Any] else {
             throw WrapperError.parametersNotCorrect(
-                description: "'headers' must be an object of containing String: String")
+                description: "'headers' must be an object of String: String")
```
Comment (szymon-onewelcome) [inline ios/Handlers/ResourcesHandler.swift:92]: You are casting to `[String: Any]` not `[String: String]`

File: ios/RNOneginiSdk.swift [hunk 2]
```diff
@@ -308,8 +308,14 @@ class RNOneginiSdk: RCTEventEmitter, ConnectorToRNBridgeProtocol {
...
         }
 
-        let completion = makeCompletionHandler(resolver: resolve, rejecter: reject, resolveWithNil: false)
-        bridgeConnector.toResourceHandler.resourceRequest(resourceType, details, completion)
+        bridgeConnector.toResourceHandler.resourceRequest(resourceType, details) { result in
+            switch result {
+            case .success(let response):
+                resolve(response.asDictionary)
+            case .failure(let error):
+                self.rejectWithError(reject, error)
```
Comment (szymon-onewelcome) [inline ios/RNOneginiSdk.swift:316]: Are you sure it can be strong reference here?

File: ios/Handlers/ResourcesHandler.swift [hunk 3]
```diff
@@ -17,18 +17,42 @@ enum ResourceRequestType: String {
     }
 }
 
+struct WrapperResourceResponse {
+    var body: String
+    var headers: [AnyHashable: String]
+    var ok: Bool
+    var status: Int
+    init (_ response: ResourceResponse) {
```
Comment (szymon-onewelcome) [inline ios/Handlers/ResourcesHandler.swift:25]: Add extra new line before this line. And remove space: `init(`

**Review by szymon-onewelcome — CHANGES_REQUESTED** [review]

File: ios/Handlers/ResourcesHandler.swift [hunk 4]
```diff
@@ -17,18 +17,42 @@ enum ResourceRequestType: String {
...
     private let deviceClient = SharedDeviceClient.instance
     private let userClient = SharedUserClient.instance
     
-    func resourceRequest(_ type: ResourceRequestType, _ details: Dictionary<String, Any?>, _ completion: @escaping (String?, Error?) -> Void) {
+    func resourceRequest(_ type: ResourceRequestType, _ details: Dictionary<String, Any?>, _ completion: @escaping (Result<WrapperResourceResponse, Error>) -> Void) {
 
         let completionHandler = { (_ response: ResourceResponse?,_ error: Error?) in
             if let error = error {
-                completion(nil, error)
+                guard let response = response else {
```
[the remark this answers is not in the document]
Comment (MichielVrins) [reply ios/Handlers/ResourcesHandler.swift:45]: This does not compile for me, is this for a new version of swift?

File: ios/RNOneginiSdk.swift (hunk 2 above)
Comment (MichielVrins) [reply ios/RNOneginiSdk.swift:316]: yes, as this is the entry point of the RN wrapper that should be fine.

File: ios/Handlers/ResourcesHandler.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (szymon-onewelcome) [reply ios/Handlers/ResourcesHandler.swift:47]: Yep, don't bother.

**Review by szymon-onewelcome — APPROVED** [review]

**Pull request merged**